docs: add new tutorial unit five (#3323)
refinedev/refine: documentation/docs/tutorial/4-adding-crud-pages/antd/add-list-page.md

import {
    Layout,
    ReadyPage,
    notificationProvider,
    ErrorComponent,
} from "@pankod/refine-antd";
import routerProvider from "@pankod/refine-react-router-v6";
import dataProvider from "@pankod/refine-simple-rest";
import { AntdInferencer } from "@pankod/refine-inferencer/antd";

import "@pankod/refine-antd/dist/reset.css";

const App: React.FC = () => {
    return (
        <Refine
            routerProvider={routerProvider}
            dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
            Layout={Layout}
            ReadyPage={ReadyPage}
            notificationProvider={notificationProvider}
            catchAll={<ErrorComponent />}
            resources={[
                {
                    name: "posts",
                    list: AntdInferencer,
                    show: AntdInferencer,
                    create: AntdInferencer,
                    edit: AntdInferencer,
                },
            ]}
        />
    );
};

render(<App />);
```

Instead of coding the list page component from scratch, Inferencer've created the required code base on API response, so that we can customize.

## Understanding the List Component

> 

Reviewer: You add similar sentence to other pages

    -   `dataIndex` prop is used to specify the field of the record that will be rendered in this column.
    -   `render` prop is used to render custom content in the column. In the auto-generated list page code, Inferencer used the `render` prop to render the `description` field using `<MarkdownField/>` component.

        [Refer to the `<MarkdownField/>` and other field components documentation for more information &#8594](/docs/api-reference/antd/components/fields/markdown/)

-   `<EditButton/>` and `<ShowButton/>` are **refine** components that are used to navigate to the edit and show pages of the record.

    [Refer to the `<EditButton/>` documentation for more information &#8594](/docs/api-reference/antd/components/buttons/edit-button/)

    [Refer to the `<ShowButton/>` documentation for more information &#8594](/docs/api-reference/antd/components/buttons/show-button/)

-   `<DeleteButton/>` is a **refine** component that is used to delete the record.

    [Refer to the `<ShowButton/>` documentation for more information &#8594](/docs/api-reference/antd/components/buttons/delete-button/)

### Handling Relationships

> Each product includes the `category` field. But, it only includes the `id` of the category. This is a foreign key that points to the `categories` resource. In the `categories` resource, we can see the `title` field. So, in order to display the category `title` in the table, we can use the `useMany` hook provided by **refine**. This hook allows us to fetch data for multiple records in a single request by providing the `id`'s of the related records. In this case, we need to provide the `id`'s of the products' categories. It is particularly useful when we need to fetch related data for multiple records.

Reviewer: ```suggestion
Each product includes the `category` field which has `id` property. This is a foreign key that points to the `categories` resource which is different than "product" resource.  
 There is a `title` field In the `categories` resource. To display the category `title` in the table, we can use the `useMany` hook provided by **refine**.   
This hook allows us to fetch data for multiple records in a single request by providing the `id`'s of the related records. In this case, we need to provide the `id`'s of the products categories. It is particularly useful when we need to fetch related data for multiple records.
```

                            languageData?.data?.find(
                                (item) => item.id === value,
                            )?.title
                        )
                    }
                />
                <Table.Column
                    title="Actions"
                    dataIndex="actions"
                    render={(_, record: BaseRecord) => (
                        <Space>
                            <EditButton
                                hideText
                                size="small"
                                recordItemId={record.id}
                            />
                            <ShowButton
                                hideText
                                size="small"
                                recordItemId={record.id}
                            />
                            <DeleteButton
                                hideText
                                size="small"
                                recordItemId={record.id}
                            />
                        </Space>
                    )}
                />
            </Table>
        </List>
    );
};
```

> Instead of starting from scratch, Inferencer've created the list page code that we can customize.

Reviewer: ```suggestion
Instead of coding the list page component from scratch, Inferencer've created the required code base on AP response, so that we can customize.
```

Reviewer: Mayba
